src/rebalancers/Rebalancer.sol

```diff
@@ -0,0 +1,492 @@
+                revert OnlyPool();
+            }
+        }
+
+        if (amount0Delta > 0) {
+            ERC20(token0).safeTransfer(msg.sender, uint256(amount0Delta));
+        } else if (amount1Delta > 0) {
+            ERC20(token1).safeTransfer(msg.sender, uint256(amount1Delta));
+        }
+    }
+
+    /* ///////////////////////////////////////////////////////////////
+                    PUBLIC POSITION VIEW FUNCTIONS
+    /////////////////////////////////////////////////////////////// */
+
+    /**
+     * @notice returns if the pool of a Liquidity Position is unbalanced.
+     * @param position Struct with the position data.
+     * @return isPoolUnbalanced_ Bool indicating if the pool is unbalanced.
+     */
+    function isPoolUnbalanced(PositionState memory position) public pure returns (bool isPoolUnbalanced_) {
+        // Check if current priceX96 of the Pool is within accepted tolerance of the calculated trusted priceX96.
+        isPoolUnbalanced_ = position.sqrtPriceX96 <= position.lowerBoundSqrtPriceX96
+            || position.sqrtPriceX96 >= position.upperBoundSqrtPriceX96;
+    }
+
+    /**
+     * @notice Fetches all required position data from external contracts.
+     * @param positionManager The contract address of the Position Manager.
+     * @param id The id of the Liquidity Position.
+     * @param tickLower The lower tick of the newly minted position.
+     * @param tickUpper The upper tick of the newly minted position.
+     * @param initiator The address of the initiator.
+     * @return position Struct with the position data.
+     */
+    function getPositionState(address positionManager, uint256 id, int24 tickLower, int24 tickUpper, address initiator)
+        public
+        view
+        returns (PositionState memory position)
+    {
+        // Get data of the Liquidity Position.
+        (int24 tickCurrent, int24 tickRange) = positionManager == address(UniswapV3Logic.POSITION_MANAGER)
+            ? UniswapV3Logic._getPositionState(position, id, tickLower == tickUpper)
+            // Logic holds for both Slipstream and staked Slipstream positions.
+            : SlipstreamLogic._getPositionState(position, id);
+
+        if (positionManager == address(StakedSlipstreamLogic.POSITION_MANAGER)) {
+            StakedSlipstreamLogic._getPositionState(position);
+        }
+
+        // Store the new ticks for the rebalance
+        if (tickLower == tickUpper) {
+            // Round current tick down to a tick that is a multiple of the tick spacing (can be initialised).
+            // ToDo: handle TICK_MAX and TICK_MIN
+            tickCurrent = tickCurrent / position.tickSpacing * position.tickSpacing;
+            (position.tickLower, position.tickUpper) = tickRange > position.tickSpacing
+                ? (tickCurrent - tickRange / 2, tickCurrent + tickRange / 2)
+                : (tickCurrent, tickCurrent + tickRange);
```

Comment:
just for info, this is arbitrary choice to do + or - tickRange here ?

Reply:
It is the smallest feasible position in this case.
And if we go from tickCurrent to (tickCurrent + tickRange) the new position is in range.